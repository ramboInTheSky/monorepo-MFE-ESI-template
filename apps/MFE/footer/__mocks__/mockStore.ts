import configureMockStore from "redux-mock-store"
import {State} from "../src/ducks"

export const mockText = {
    defaultCopyrightText: `Amido Retail Ltd. All rights reserved.`,
}

export const mockState: State = {
    user: {accountFirstName: "", loggedIn: false},
    request: {headers: {}, url: "/", isInternationalCountry: false},
    textAlignment: "ltr",
    data: {
        id: "",
        name: "",
        realm: "",
        territory: "",
        language: "",
        regions: [
            {
                type: "",
                title: "",
                accessibilityTitle: "",
                subRegions: [
                    {
                        title: "",
                        accessibilityTitle: "",
                        elements: [
                            {
                                url: "",
                                openInNewWindow: false,
                                name: "",
                                type: "",
                                icon: "",
                                text: "",
                                accessibilityText: "",
                                tooltip: "",
                                accessibilityTooltip: "",
                                description: "",
                                accessibilityDescription: "",
                            },
                        ],
                    },
                ],
            },
        ],
    },
    languages: {
        currentLanguageText: "English",
        altLanguageName: "ar",
        altLanguageUrl: "/ar",
        currentLanguageName: "en",
        direction: "ltr",
        siteUrl: "http://amido.com",
    },
    text: mockText,
    settings: {variant: "default"},
}

const mockConfigureStore = configureMockStore()
const mockStore = mockConfigureStore(mockState)

export default mockStore
