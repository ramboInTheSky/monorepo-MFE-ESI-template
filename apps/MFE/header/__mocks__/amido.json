{
  "id": "33d0153c-ed8b-4a65-b63c-5ad5edddd187",
  "name": "Default",
  "realm": "amido",
  "territory": "GB",
  "language": "en",
  "regions": [
    {
      "type": "Brand",
      "title": "Brand",
      "accessibilityTitle": "Brand",
      "elements": [
        {
          "elements": [],
          "type": "link",
          "url": "/",
          "openInNewWindow": false,
          "icon": "/icons/header/brand/amido/logo.svg",
          "text": "Amido Logo",
          "accessibilityText": "Amido Icon",
          "tooltip": "Amido Icon",
          "wideModeIcon": "/icons/header/brand/amido/logo.svg",
          "narrowModeIcon": "/icons/header/brand/amido/logo.svg"
        }
      ]
    },
    {
      "type": "QuickLinks",
      "title": "QuickLinks",
      "accessibilityTitle": "QuickLinks",
      "elements": [
        {
          "elements": [
            {
              "type": "MyAccountLoggedIn",
              "url": "/secure/accounts/transfer",
              "openInNewWindow": false,
              "icon": "/icons/header/quicklinks/amido/my-account.svg",
              "text": "<username>",
              "accessibilityText": "<username>",
              "tooltip": "<username>'s account",
              "wideModeIcon": "/icons/header/quicklinks/amido/my-account_desktop.svg",
              "narrowModeIcon": "/icons/header/quicklinks/amido/my-account.svg"
            },
            {
              "type": "MyAccountLoggedOut",
              "url": "/secure/accounts/transfer",
              "openInNewWindow": false,
              "icon": "/icons/header/quicklinks/amido/my-account.svg",
              "text": "My Account",
              "accessibilityText": "Sign in to view account details",
              "tooltip": "Sign in to view account details",
              "wideModeIcon": "/icons/header/quicklinks/amido/my-account_desktop.svg",
              "narrowModeIcon": "/icons/header/quicklinks/amido/my-account.svg"
            }
          ],
          "type": "MyAccount",
          "url": null,
          "openInNewWindow": true,
          "icon": null,
          "text": "MyAccount",
          "accessibilityText": "MyAccount",
          "tooltip": null,
          "wideModeIcon": null,
          "narrowModeIcon": null
        },
        {
          "elements": [],
          "type": "link",
          "url": "/help",
          "openInNewWindow": true,
          "icon": null,
          "text": "Help",
          "accessibilityText": "Help",
          "tooltip": "Help - FAQ's, how to shop online, sizing guides and more",
          "wideModeIcon": null,
          "narrowModeIcon": null
        },
        {
          "elements": [],
          "type": "link",
          "url": "/stores/",
          "openInNewWindow": false,
          "icon": null,
          "text": "Store Locator",
          "accessibilityText": "Store Locator",
          "tooltip": "Find your nearest Amido Store",
          "wideModeIcon": null,
          "narrowModeIcon": null
        }
      ]
    }
  ]
}
