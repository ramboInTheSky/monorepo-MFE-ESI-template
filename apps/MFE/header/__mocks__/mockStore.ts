import configureMockStore from "redux-mock-store"
import thunk from "redux-thunk"
import {mockColors} from "@monorepo/themes"
import {State} from "../src/ducks"
import apiData from "./amido.json"
import {ShoppingBag, ChargesAndIncentive, Item} from "../src/models/shoppingbag"

export const mockStyles = {
    sm: {
        width: "5.375rem",
        height: "2.75rem",
        padding: "0.625rem 0",
        borderRadius: "0",
        border: "0",
        order: "1",
    },
    xs: {
        width: "5.375rem",
        height: "2.75rem",
        padding: "0.625rem 0",
        borderRadius: "0",
        border: "0",
        order: "1",
    },
    md: {
        width: "5.375rem",
        height: "2.75rem",
        padding: "0.625rem 0",
        borderRadius: "0",
        border: "0",
        order: "1",
        margin: "10rem",
        decoration: "none",
    },
    lg: {
        width: "5.375rem",
        height: "2.75rem",
        padding: "0.625rem 0",
        borderRadius: "0",
        border: "0",
        order: "1",
    },
    xl: {
        width: "5.375rem",
        height: "2.75rem",
        padding: "0.625rem 0",
        borderRadius: "0",
        border: "0",
        order: "1",
    },
}

export const mockMyAccountStyles = {
    ...mockStyles,
    SmallIcon: mockStyles,
    SmallIconContainer: mockStyles,
    IconOnly: mockStyles,
    IconContainer: mockStyles,
    ToolTip: mockStyles,
    Divider: mockStyles,
}

export const mockCountrySelectorStyles = {
    ...mockStyles,
    Container: mockStyles,
}

const mockShoppingBagStyles = {
    ...mockStyles,
    productImage: mockStyles,
    itemsContainer: mockStyles,
}

const mockFavouritesStyles = {
    ...mockStyles,
    img: mockStyles,
}

export const mockRealmStyles = {
    CTA: mockStyles,
    UpperHeader: mockStyles,
    Meganav: mockStyles,
    PrimaryNav: mockStyles,
    Brand: mockStyles,
    SearchBox: mockStyles,
    Checkout: mockStyles,
    Quicklinks: mockStyles,
    SearchBoxDivider: mockStyles,
    SearchBoxIcon: mockStyles,
    Favourites: mockFavouritesStyles,
    CountrySelector: mockCountrySelectorStyles,
    ShoppingBag: mockShoppingBagStyles,
    MyAccount: mockMyAccountStyles,
    AutoCompleteButton: mockStyles,
    DrawerContainer: mockStyles,
}

const discount = 10
export const mockShoppingBag = {
    itemCount: 2,
    loading: false,
    bag: {
        FirstName: "cypress-account",
        Description: "Excluding UK Standard Delivery (Normally £3.99)",
        MultiBuyDiscount: 1,
        FinalOrderValue: 89,
        FinalOrderValueFormatted: "£89.00",
        MultiBuyDiscountFormatted: "£1.00",
        ChargesAndIncentives: [
            {
                AdditionalAmount: -discount,
                AdditionalAmountFormatted: `-£${discount.toFixed(2)}`,
                AdditionalCode: "UKCO1",
                MaximumOrderValue: 0,
                MinimumOrderValue: 15,
                OfferShortDescription: "Welcome Offer",
                OfferTypeDescription: "Welcome Offer",
                Type: "PRO",
            } as ChargesAndIncentive,
        ],
        ItemCount: 2,
        Items: [
            {
                ItemID: 1,
                ItemNumber: "436441",
                OptionNo: "04",
                SizeDescription: "10 S",
                Price: "45.00",
                Quantity: 10,
                StockMessage: "Delayed",
                Description: "Black Lift, Slim And Shape Skinny Jeans",
                AlternativeDescription: "Alternative - Black Lift, Slim And Shape Skinny Jeans",
                CistStoreName: "",
                Url: "/path1",
            } as Item,
            {
                ItemID: 2,
                ItemNumber: "436642",
                OptionNo: "04",
                SizeDescription: "15 L",
                Price: "45.00",
                Quantity: 5,
                StockMessage: "In Stock",
                Description: "Black Lift, Slim And Shape Skinny Jeans 2",
                AlternativeDescription: "Alternative - Black Lift, Slim And Shape Skinny Jeans 2",
                CistStoreName: "LEICESTER - HIGH CROSS SHOPPING CENTRE",
                Url: "/path2",
            } as Item,
        ],
    } as ShoppingBag,
}

export const mockRecents = {
    queryIds: ["celio", "gucci", "prada", "&", "#", "+"],
}

export const mockAutoComplete: any = {
    isLoading: false,
    q: "socks",
    suggestions: [
        {
            dq: "girls socks",
            q: "girls socks",
        },
        {
            dq: "boys socks",
            q: "boys socks",
        },
    ],
    products: [],
    numFound: 2,
    parameters: {
        accountId: "abc",
        domainKey: "Amido",
    },
}

export const mockLanguages = {
    currentLanguageText: "English",
    altLanguageName: "ar",
    altLanguageUrl: "/ar",
    currentLanguageName: "en",
    direction: "ltr",
    siteUrl: "http://amido.com",
}

export const mockSearch = {
    typedCharacters: "",
    showAutocomplete: false,
    showRecentSearch: false,
}

export const mockText = {
    recentSearches: {
        enriched: {
            mobileHeader: "RECENT SEARCHES",
            desktopHeader: "Your recent searches: ",
        },
        simple: {
            mobileHeader: "RECENT SEARCHES",
            desktopHeader: "RECENT SEARCHES",
        },
    },
    searchBox: {
        label: "Search",
        bigPlaceholder: "Search product",
        smallPlaceholder: "Search...",
        clearButton: {
            alt: "Clear search button",
        },
        closeButton: {
            text: "Close",
        },
    },
    myAccount: {
        title: "View Account Summary",
        buttonText: "SIGN OUT",
        altText: "My account icon",
        tooltipAltText: "icon",
        notText: "Not",
    },
    checkout: {
        text: "Checkout",
    },
    autoComplete: {
        loadingText: "Loading...",
        noResults: "No results found for ",
        headerText: "TOP MATCHES",
        searchLinktext: "See all results",
        enrich: {
            termTitle: "Top results for ",
        },
    },
    drawer: {
        closeIconText: "Close icon",
    },
    favourites: {
        altText: "Favourites icon",
    },
    miniShoppingBag: {
        emptyBagText: "Your shopping bag is empty",
        totalText: "Total",
        promotionalText: "Promotional Savings",
        viewAndEditBag: "VIEW/EDIT BAG",
        sizeText: "Size:",
        qtyText: "Qty:",
        at: "at",
        negative: "-",
        itemText: "Item",
        itemsText: "Items",
        inBagText: "In Bag",
        salePrice: "Sale Price",
    },
    shoppingBag: {
        altText: "Shopping bag icon",
        youhaveText: "You have",
        itemText: "item",
        itemsText: "items",
        inYourBagText: "in your bag",
    },
    countryLangSelector: {
        headerTitle: "CHANGE COUNTRY",
        locationText: "Location",
        languageText: "Language",
        closeText: "Close",
        stayText: "Stay on Amido.co.uk",
        shopNowText: "Shop now",
        ROWText: "Can't find the country you're looking for?",
        ROWLinkText: "View All Countries",
    },
    countryChangeModal: {
        confirm: "CONTINUE",
        cancel: "CANCEL",
        bodyTextPt1: "Are you sure you want to navigate away from this site?",
        bodyTextPt2: "If you navigate away from this site you will leave your shopping bag and it’s contents.",
        titleText: "Confirm country change",
    },
    cookiePolicy: {
        title: "Cookie Policy",
        body: "We use cookies to provide you with the best posible experience. By continuing to use our site, you agree to our use of cookies.",
        link: "Find out more",
        linkText: "about managing your cookie settings.",
        closeIconAltText: "Close",
    },
    saleBagWarning: {
        title: "Careful",
        textOne: "You could lose your sale bag.",
        textTwo: "Would you like to return to our Sale Site?",
        primaryButtonText: "YES, RETURN TO THE SALE",
        secondaryButtonText: "NO, THANK YOU",
    },
    skipToMainContent: "Skip to main content",
    closeIconAltText: "Close",
}

export const mockState: State = {
    settings: {
        "userConsentManagement.enabled": false,
        variant: "default",
    },
    user: {firstName: "", loggedIn: false, userUpdated: false},
    autocomplete: mockAutoComplete,
    recents: mockRecents,
    shoppingBag: mockShoppingBag,
    search: mockSearch,
    template: "default",
    request: {
        headers: {"x-monorepo-territory": "gb"},
        url: "/",
        siteUrl: "fakeamido.com",
        timeMachineDate: "20201215",
        isInternationalCountry: false,
        fullTerritoryName: "United Kingdom",
        geolocationBaseUrl: "",
        geolocationVersion: 0,
        bloomReachCachingCookieList: "",
        bloomReachCachingEnabled: false,
        useTimeMachineCookie: false,
        currencyCode: "EUR",
        bloomreachGroupLocation: "southernhemisphere",
        bloomreachDomainKey: "GB",
        monetateSDK: false,
        accountMonetateSDK: "",
        cookieDomain: ".amido.com",
        showSaleWarningBag: false,
    },
    languages: mockLanguages,
    data: {...apiData, language: ""},
    textAlignment: "ltr",
    text: mockText,
    features: {
        SearchBar: {
            Value: "SimpleSearch",
            RecentSearch: {
                MaxItems: 12,
            },
            Autocomplete: {
                MaxItems: 10,
                ProductsMaxItems: {
                    xs: 8,
                    sm: 8,
                    md: 4,
                    lg: 4,
                    xl: 5,
                },
            },
        },
    },
    favourites: {enableFavourites: true, hasFavourites: true},
    countrySelector: {
        showCountrySelector: false,
        isActive: true,
        countriesList: null,
        selectedCountry: null,
        selectedLanguage: null,
        defaultCountryCode: null,
        showOverlay: true,
        showBFPOFlag: false,
        loaded: false,
        requestedCountryChange: false,
        countrySelectorEndpoint: "",
    },
}

export const mockTextAlignment = {Value: "ltr"}

export const mockTheme = {styles: mockRealmStyles, colours: mockColors, direction: "ltr"}
export const mockConfigureStore = configureMockStore([thunk])
const mockStore = mockConfigureStore(mockState)

export default mockStore
