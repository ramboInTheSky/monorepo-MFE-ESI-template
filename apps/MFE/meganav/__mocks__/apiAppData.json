{
  "id": "d62d81d8-3d09-41a1-b427-ac1a162a3976",
  "title": "Home",
  "target": "amido.com/home",
  "viewType": "primary-navbar",
  "items": [
    {
      "title": "New In",
      "target": "/new-in",
      "path": "New In",
      "excludeFrom": ["NarrowView", "WideView"]
    },
    {
      "title": "Girls",
      "target": "/girls",
      "path": "Girls",
      "excludeFrom": ["NarrowView"]
    },
    {
      "title": "Lingerie",
      "target": "/lingerie-shop",
      "path": "Lingerie",
      "excludeFrom": ["WideView"]
    },
    {
      "title": "Women",
      "target": "/women",
      "path": "Women",
      "linkColour": "#008001",
      "fontWeight": "700",
      "fontFamily": "Century Gothic",
      "excludeFrom": []
    },
    {
      "title": "Home",
      "target": "/homeware",
      "path": "Home"
    },
    {
      "title": "Boys",
      "target": "/boys",
      "path": "Boys"
    },
    {
      "title": "Men",
      "target": "/men",
      "path": "Men"
    },
    {
      "title": "Shoes",
      "target": "/shoes",
      "path": "Shoes"
    },
    {
      "title": "Beauty",
      "target": "/beauty",
      "path": "Beauty"
    },
    {
      "title": "Brands",
      "target": "/branded",
      "path": "Brands"
    },
    {
      "title": "Sports",
      "target": "/sports",
      "path": "Sports"
    },
    {
      "title": "Gifts & Flowers",
      "target": "/gifts-flowers",
      "path": "Gifts & Flowers"
    },
    {
      "title": "Clearance",
      "target": "/clearance",
      "path": "Clearance"
    }
  ]
}
