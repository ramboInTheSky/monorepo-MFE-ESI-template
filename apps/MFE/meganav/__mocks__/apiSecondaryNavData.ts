import {SecondaryNav} from "../src/models/secondary-nav"

const secondaryNavMockData: SecondaryNav = {
    id: "3df60188-7db2-433c-b6c2-7b5034538fb2",
    title: "GIRLS",
    target: "/girls",
    viewType: "secondary-navigation",
    realm: "amido",
    territory: "GB",
    language: "en",
    items: [
        {
            title: "GIRLS",
            type: "tab",
            items: [
                {
                    title: "Column 1",
                    type: "column",
                    items: [
                        {
                            title: "",
                            type: "category",
                            icon: null,
                            items: [
                                {
                                    title: "New In",
                                    type: "link",
                                    icon: null,
                                    target: "/shop/gender-newborngirls-gender-newbornunisex-gender-oldergirls-gender-youngergirls/feat-newin",
                                    linkColour: "",
                                    fontWeight: "",
                                    fontFamily: null,
                                },
                            ],
                            linkColour: "",
                            fontWeight: "",
                            fontFamily: null,
                        },
                    ],
                },
            ],
            missions: null,
        },
        {
            title: "exclude wide tab",
            type: "tab",
            items: [],
            missions: null,
            excludeFrom: ["WideView"],
        },
        {
            title: "exclude narrow tab",
            type: "tab",
            items: [],
            missions: null,
            excludeFrom: ["NarrowView"],
        },
    ],
    banner: null,
}

export default secondaryNavMockData
