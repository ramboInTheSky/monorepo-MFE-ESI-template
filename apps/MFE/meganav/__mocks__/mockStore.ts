import configureMockStore from "redux-mock-store"
import thunk from "redux-thunk"
import {mockColors} from "@monorepo/themes"
import {State} from "../src/ducks"
import PrimaryNavData from "./apiAppData.json"
import SecondaryNavData from "./apiSecondaryNavData"
import {SettingsModel} from "../src/ducks/settings"
import {MissionViewportDimensions} from "../src/models/dimensions"
import {CompositionSettingsDuckState} from "../src/ducks/compositionSettings"

export const mockDimensions = {
    sm: {
        width: "5.375rem",
        height: "2.75rem",
        padding: "0.625rem 0",
        borderRadius: "0",
        border: "0",
        order: "1",
        justifyContent: "center",
    },
    xs: {
        width: "5.375rem",
        height: "2.75rem",
        padding: "0.625rem 0",
        borderRadius: "0",
        border: "0",
        order: "1",
        justifyContent: "center",
    },
    md: {
        width: "5.375rem",
        height: "2.75rem",
        padding: "0.625rem 0",
        borderRadius: "0",
        border: "0",
        order: "1",
        justifyContent: "center",
    },
    lg: {
        width: "5.375rem",
        height: "2.75rem",
        padding: "0.625rem 0",
        borderRadius: "0",
        border: "0",
        order: "1",
        justifyContent: "center",
    },
    xl: {
        width: "5.375rem",
        height: "2.75rem",
        padding: "0.625rem 0",
        borderRadius: "0",
        border: "0",
        order: "1",
        justifyContent: "center",
    },
}

const missionsMockDimensions: MissionViewportDimensions = {
    Container: mockDimensions,
    Item: mockDimensions,
}

export const mockRealmDimensions = {
    UpperHeader: mockDimensions,
    PrimaryNav: mockDimensions,
    SecondaryNav: mockDimensions,
    SnailItem: mockDimensions,
    Missions: missionsMockDimensions,
}

export const mockLanguages = {
    currentLanguageText: "English",
    altLanguageName: "ar",
    altLanguageUrl: "/ar",
    currentLanguageName: "en",
    direction: "ltr",
    siteUrl: "http://amido.com",
}

export const mockTheme = {
    dimensions: mockRealmDimensions,
    colours: mockColors,
}

export const mockAccordionActivity = {}
export const mockTabsActivity = {
    men: 0,
}

export const mockMissions = {
    title: "sample-missions-title",
    categoryLink: {title: "categoryLinkTitle", target: "/categoryLinkTarget"},
    items: [],
    noOfColumns: 6,
}

export const mockPrimarynav = {
    activeDepartment: "men",
    active: false,
    activeDepartmentIndex: -1,
    items: PrimaryNavData.items,
    isInPrimaryNav: false,
    config: {
        version: "v1.0.0",
        country: "mx",
    },
}

export const mockBanner = {
    imageUrl: "/sampleimageurl",
    target: "/sampletarget",
}

export const mockText = {
    arrowIconUrlAltText: "Menu expand icon",
    bannerAltText: "Banner image",
    chevronIconAltText: "Chevron icon",
    drawerIconAltText: "Drawer close icon",
}

export const mockSecondarynav = {
    departmentIds: ["men"],
    catalogues: {
        men: SecondaryNavData,
    },
    isPending: false,
    isInSecondaryNav: false,
    isImagePlaceholderEnabled: false,
    config: {
        version: "v1.0.0",
        country: "mx",
    },
}

export const mockSettingsModel: SettingsModel = {
    template: "default",
}

export const mockCompositionSettings: CompositionSettingsDuckState = {
    showSecondaryNavArrow: false,
}

export const mockState: State = {
    primarynav: mockPrimarynav as any,
    secondarynav: mockSecondarynav,
    accordionActivity: mockAccordionActivity,
    tabsActivity: mockTabsActivity,
    settings: mockSettingsModel,
    request: {
        headers: {"x-monorepo-territory": "gb"},
        url: "/",
        siteUrl: "",
    },
    textAlignment: "ltr",
    text: mockText,
    compositionSettings: mockCompositionSettings,
}

const mockConfigureStore = configureMockStore([thunk])
const mockStore = mockConfigureStore(mockState)

export default mockStore
