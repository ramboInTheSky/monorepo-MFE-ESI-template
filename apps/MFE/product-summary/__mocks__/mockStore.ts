import configureMockStore from "redux-mock-store"
import {mockColors} from "@monorepo/themes"
import State from "../src/models/state"
import {Fits, ProductType, SofaType} from "../src/config/constants"
import text from "./default-text.json"

export const mockState: State = {
    text,
    textAlignment: "ltr",
    productSummary: {
        summaryData: {
            title: "My Product",
            id: "00000",
            brand: "nike",
            department: "menswear",
            fit: "tall",
            currencyCode: "GBP",
            baseUrl: "http://test.co.uk",
            imageCdnUrl: "http://testcdn.co.uk",
            productImageUrlPart: "search",
            showNewIn: false,
            saleSash: null,
            saleSashPosition: null,
            colourways: [
                {
                    title: "White",
                    fits: [Fits.Tall],
                    id: "99434296",
                    colour: "White",
                    url: "g5990s21/845632#845632",
                    selected: true,
                    price: "£123",
                    salePrice: null,
                    wasPrice: null,
                    minPrice: "£123",
                    overallStarRating: 0,
                    colourChipImage: "http://testcdn.co.uk/AltItemSwatch/21x21/99434296.jpg",
                    mainImage: "http://testcdn.co.uk/search/224x336/99434296.jpg",
                    personalisedType: "Test",
                },
                {
                    title: "Black",
                    fits: [Fits.Tall, Fits.Petite],
                    id: "434296",
                    colour: "Black",
                    url: "g5990s21/845632#845632",
                    selected: false,
                    price: "£99",
                    salePrice: null,
                    wasPrice: null,
                    minPrice: "£99",
                    overallStarRating: 4,
                    colourChipImage: "http://testcdn.co.uk/AltItemSwatch/21x21/434296.jpg",
                    mainImage: "http://testcdn.co.uk/search/224x336/434296.jpg",
                    personalisedType: "Test",
                },
            ],
            type: ProductType,
        },
        canLoadVisibleColourways: false,
        hasLoadedVisibleColourways: false,
        isFav: false,
        enableFav: false,
        enableReviewStars: false,
        showFavErrorToolTip: null,
        enabledSearchDesc: false,
        isLoadingFav: false,
        animateFavouriteIcon: false,
        enableBrandDisplay: false,
    },
    lazyload: {
        colourchips: false,
        productImages: false,
        fitIcons: false,
    },
    request: {
        headers: {"x-monorepo-language": "en"},
        isEnglishLang: true,
    },
}

export const mockStateSofa: State = {
    ...mockState,
    productSummary: {
        ...mockState.productSummary,
        summaryData: {
            title: "Jackson",
            id: "jackson_82_262919",
            brand: "",
            department: "homeware",
            fit: "",
            currencyCode: "GBP",
            baseUrl: "http://test.co.uk",
            imageCdnUrl: "http://testcdn.co.uk",
            productImageUrlPart: "search",
            showNewIn: false,
            saleSash: null,
            saleSashPosition: null,
            type: SofaType,
            colourways: [
                {
                    title: "Sofa 1",
                    fits: [],
                    id: "jackson_82_262919",
                    colour: "Chunky Weave/Dove",
                    url: "/homeware/sofas-chairs/jackson/size-power-recliner-large-sofa-fabric-chunky-weave-dove",
                    selected: true,
                    price: "£775 - £1,350",
                    salePrice: null,
                    wasPrice: null,
                    minPrice: "£775",
                    mainImage: "http://testcdn.co.uk/262919_82.jpg",
                    colourChipImage: "http://testcdn.co.uk/SofaSwatchView/chunkyweave_dove.jpg",
                    overallStarRating: 0,
                    personalisedType: "",
                },
                {
                    title: "Sofa 2",
                    fits: [],
                    id: "jackson_82_202546",
                    colour: "Monza Faux Leather/Charcoal",
                    url: "/homeware/sofas-chairs/jackson/size-power-recliner-large-sofa-fabric-monza-faux-leather-charcoal",
                    selected: false,
                    price: "£775 - £1,350",
                    salePrice: null,
                    wasPrice: null,
                    minPrice: "£775",
                    mainImage: "http://testcdn.co.uk/202546_82.jpg",
                    colourChipImage: "http://testcdn.co.uk/SofaSwatchView/monzafauxleather_charcoal.jpg",
                    overallStarRating: 0,
                    personalisedType: "",
                },
                {
                    title: "Sofa 3",
                    fits: [],
                    id: "jackson_82_286382",
                    colour: "Matt Velvet/Navy Jackson",
                    url: "/homeware/sofas-chairs/jackson/size-power-recliner-large-sofa-fabric-matt-velvet-navy",
                    selected: false,
                    price: "£775 - £1,350",
                    salePrice: null,
                    wasPrice: null,
                    minPrice: "£775",
                    mainImage: "http://testcdn.co.uk/286382_82.jpg",
                    colourChipImage: "http://testcdn.co.uk/SofaSwatchView/mattvelvet_navy.jpg",
                    overallStarRating: 0,
                    personalisedType: "",
                },
            ],
        },
    },
}

export const mockTextAlignment = {Value: "ltr"}

export const mockTheme = {colours: mockColors, direction: "ltr", enablingCenteredContent: false}
export const centralisedMockTheme = {colours: mockColors, direction: "ltr", enablingCenteredContent: true}
const mockConfigureStore = configureMockStore()
const mockStore = mockConfigureStore(mockState)

export default mockStore
