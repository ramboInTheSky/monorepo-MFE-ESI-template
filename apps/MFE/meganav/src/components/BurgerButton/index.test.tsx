import React from "react"
import {render} from "@testing-library/react"
import {SCThemeProvider} from "@monorepo/theme-provider"

import {BurgerButton} from "."
import {mockTheme} from "../../../__mocks__/mockStore"

describe("Components - BurgerButton: ", () => {
    it("should match the snapshot", () => {
        const {asFragment} = render(
            <SCThemeProvider theme={mockTheme}>
                <BurgerButton />
            </SCThemeProvider>,
        )
        expect(asFragment()).toMatchSnapshot()
    })
})
