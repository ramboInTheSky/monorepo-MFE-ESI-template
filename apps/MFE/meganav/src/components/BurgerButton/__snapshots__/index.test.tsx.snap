// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components - BurgerButton:  should match the snapshot 1`] = `
<DocumentFragment>
  <button
    aria-label="Mobile Menu Toggle"
    class="component__Button-sc-1l1bm2s-0 itVwJG"
    type="button"
  >
    <img
      alt="burger icon"
      src="/static-content/icons/meganav/shared/menu.svg"
    />
  </button>
</DocumentFragment>
`;
